### ISSUE TITLE
Draw simple point or graphic

### ISSUE THREAD

**Issue by sindizzy** [DotSpatial/DotSpatial#1181]
DotSpatial version:  2.0

Is there a way to quickly draw a point without doing a whole featureset?

In MapWindow I used to use
```
'create a new drawing layer on map and draw a small red point as a marker
Dim hndRRcnt As Integer = mwMapMain.NewDrawing(MapWinGIS.tkDrawReferenceList.dlSpatiallyReferencedList)
mwMapMain.DrawPoint(CoordX, CoordY, 3, System.Convert.ToUInt32(RGB(0, 0, 255)))
```
So was wondering if I could do something similar in DotSpatial.

**Comment by jany-tenaj** [comment]
Take a look at the draw function of [MapFunctionMeasure](https://github.com/DotSpatial/DotSpatial/blob/master/Source/DotSpatial.Plugins.Measure/MapFunctionMeasure.cs).

As far as I know this is the only way to draw something not coming from a layer.

**Comment by sindizzy** [comment]
I think I tried drawing directly on the map but didn't work. Let me try again using the reference as a guide.

**Comment by VectorZita** [comment]
Indeed, upon drawing, the map draws everything from within its contained DrawingLayers and Layers, and then it draws whatever currently active map functions contain as instructions. From an abstract point of view, inheriting from MapFunction gives you access to the Graphics object that is drawing on the map after basic drawing has finished, so that if your MapFunction is active, whatever you paint using the Graphics object supplied in the Draw function, will be shown on the map on top of everything else contained.

Technically, all you need to draw something on the map is the Graphics object used in the Paint event. Therefore, wherever you attach a handler to the Paint event of the map, you get access to this Graphics object, and you can do drawing **on top of the map's existing contents**.
